File: IdeaProject/КИТПО/kitpo_lr_4/src/main/scala/MyTreeInt.scala
class MyTreeInt(val newType: String) extends MyTree {
  _type = newType
  override def ParseValue(str: String): Any = {
    return str.toInt
  }

  override def Clone(): MyTree = {
    return new MyTreeInt("Integer")
  }

  override def comapareTo(obj1: Any, obj2: Any): Int = {
    if (obj1.asInstanceOf[Int] < obj2.asInstanceOf[Int]) return -1
    else if (obj1.asInstanceOf[Int] == obj2.asInstanceOf[Int]) return 0
    else return 1
  }
}

File: IdeaProject/КИТПО/kitpo_lr1/src/MyTreeInt.java
public class MyTreeInt extends MyTree {
    MyTreeInt(String type) {
        super(type);
    }

    @Override
    public Object parseValue(String str) {
        return (Integer.parseInt(str));
    }

    @Override
    public int comapareTo(Object obj1, Object obj2) {
        if((Integer) obj1 == (Integer) obj2) return 0;
        else if((Integer) obj1 > (Integer) obj2) return 1;
        else  return -1;
    }
}
